src/middleware/normalize.rs

```diff
@@ -156,6 +173,10 @@ mod tests {
         let req3 = TestRequest::with_uri("//v1///something").to_srv_request();
         let res3 = normalize.call(req3).await.unwrap();
         assert!(res3.status().is_success());
+
+        let req4 = TestRequest::with_uri("/v1//something").to_srv_request();
+        let res4 = normalize.call(req4).await.unwrap();
+        assert!(res4.status().is_success());
```

Comment:
This test case would not work with the previous implementation using just string length. The path `/v1//something` will get normalized to `/v1/something/` and so its length won't change.